### PR TITLE
testcase/ta_tc/Arastorage : add ITCs for Arastorage APIs

### PR DESCRIPTION
**Pull request by arvinmittal**
Scenarios ITCs perform below tests:
itc_arastorage_db_init_deinit_p_multi_time : check iterative initialization and deinitialization database operation returns
itc_arastorage_db_print_header_n_after_deinit : Check print header API after db_deinit
itc_arastorage_db_print_tuple_n_after_deinit : Check print tuple API after db_deinit
itc_arastorage_db_print_value_n_after_deinit : Check print value API after db_deinit
itc_arastorage_cursor_get_count_n_after_deinit : Check cursor_get_count API after db_deinit
itc_arastorage_cursor_get_row_n_after_deinit : Check cursor_get_row after db_deinit
itc_arastorage_db_query_p_create_remove_relation : create relation and then remove relation
itc_arastorage_db_query_n_recreate_relation : recreate same relation which is already created
itc_arastorage_db_query_n_reremove_relation : reremove relation which is already removed
itc_arastorage_db_query_p_create_remove_multi_relation : create multiple relation and then remove multiple relation
itc_arastorage_db_query_p_create_remove_relation_multiple_time : create and remove same relation multiple time
itc_arastorage_db_get_result_message_p_all_msgs : get_resul_message for all DB_ERRORS
itc_arastorage_db_get_result_message_n : get_resul_message for specific ERROR
itc_arastorage_db_print_header_n : print header with invalid argument
itc_arastorage_db_print_tuple_n : print tuple with invelid argument
itc_arastorage_db_print_value_n : print value with invalid argument
itc_arastorage_cursor_get_count_n : get cursor count with invalid scenario
itc_arastorage_cursor_move_first_n : cursor_move_first with invalid scenario
itc_arastorage_cursor_move_next_n : cursor_move_next with invalid scenario
itc_arastorage_cursor_move_prev_n : cursor_move_prev with invalid scenario
itc_arastorage_cursor_move_last_n : cursor_move_last with invalid scenario
itc_arastorage_cursor_move_to_n : cursor_move_to with invalid scenario
itc_arastorage_cursor_is_first_row_n : cursor_is_first_row with invalid scenario
itc_arastorage_cursor_is_last_row_n : cursor_is_last_row with invalid scenario
itc_arastorage_cursor_get_attr_type_n : cursor_get_attr_type with invalid argument
itc_arastorage_cursor_get_attr_name_n : cursor_get_attr_name with invalid argument
itc_arastorage_cursor_get_attr_index_n : cursor_get_attr_index with invalid argument
itc_arastorage_cursor_get_int_value_n : cursor_get_int_value with invalid argument
itc_arastorage_cursor_get_long_value_n : cursor_get_long_value with invalid argument
itc_arastorage_cursor_get_row_p : cursor_get_row from current db cursor
itc_arastorage_cursor_get_row_n : cursor_get_row with invalid argument
itc_arastorage_cursor_get_double_value_n : cursor_get_double_value with invalid argument
itc_arastorage_cursor_get_string_value_n : cursor_get_string_value with invalid argument
itc_arastorage_db_cursor_free_p : Free allocated resources for cursor
itc_arastorage_db_cursor_free_n : Free allocated resources for cursor with NULL value

Signed-off-by: Arvin Mittal <arvin.mittal@samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

File: apps/examples/testcase/ta_tc/arastorage/itc/itc_arastorage_main.c [hunk 1]
```diff
@@ -56,127 +55,665 @@ struct arastorage_data_type_s {
...
+	ret = db_exec(g_query);
+	TC_ASSERT_EQ_CLEANUP("db_exec", ret, DB_OK, db_cleanup(rel_name); db_deinit());
+
+	memset(g_query, 0, QUERY_LENGTH);
+	sprintf(g_query, "CREATE INDEX %s.%s TYPE %s;", rel_name, g_attribute_set[0], RELATION_INDEX);
+	ret = db_exec(g_query);
+	TC_ASSERT_EQ_CLEANUP("db_exec", ret, DB_OK, db_cleanup(rel_name); db_deinit());
+
+	int index;
+	for (index = 0; index < DATA_SET_NUM; index++) {
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/ta_tc/arastorage/itc/itc_arastorage_main.c]: There are many duplcated loop code in this file, why don't you make functions for these?
insert_blah_blah() ...

File: apps/examples/testcase/ta_tc/arastorage/itc/itc_arastorage_main.c [hunk 2]
```diff
@@ -56,127 +55,665 @@ struct arastorage_data_type_s {
...
+
+	ret = db_init();
+	TC_ASSERT_EQ("db_init", ret, DB_OK);
+
+	memset(g_query, 0, QUERY_LENGTH);
+	snprintf(g_query, QUERY_LENGTH, "CREATE RELATION %s;", rel_name);
+	ret = db_exec(g_query);
+	TC_ASSERT_EQ("db_exec", ret, DB_OK);
+
+	memset(g_query, 0, QUERY_LENGTH);
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/ta_tc/arastorage/itc/itc_arastorage_main.c]: Here also create_attribute() will be fine

**Comment by Taejun-Kwon** [comment]
It is too huge to review, first, please apply my comment and then I will review again.
Thanks!

**Comment by arvinmittal** [comment]
@Taejun-Kwon 
Fixed your review comments. Please review.

**Comment by seinfra** [comment]
Code Rule Check OK.

File: apps/examples/testcase/ta_tc/arastorage/itc/itc_arastorage_main.c [hunk 3]
```diff
@@ -56,33 +55,326 @@ struct arastorage_data_type_s {
...
 };
 
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
 
-char g_query[QUERY_LENGTH];
-bool g_check;
-bool g_init;
+static char g_query[QUERY_LENGTH];
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/ta_tc/arastorage/itc/itc_arastorage_main.c]: move to above(global variables).

File: apps/examples/testcase/ta_tc/arastorage/itc/itc_arastorage_main.c [hunk 4]
```diff
@@ -56,33 +55,326 @@ struct arastorage_data_type_s {
...
+	if (g_cursor == NULL) {
+		g_flag = false;
+		printf("\ndb_query FAIL\n");
+		db_cleanup(rel_name);
+		db_deinit();
+		return;
+	}
+
+	ret = db_deinit();
+	if (ret != DB_OK) {
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/ta_tc/arastorage/itc/itc_arastorage_main.c]: this code also duplicated.
you'd better implement goto_error_exec & goto_error_query

File: apps/examples/testcase/ta_tc/arastorage/itc/itc_arastorage_main.c [hunk 5]
```diff
@@ -56,33 +55,326 @@ struct arastorage_data_type_s {
...
+* @brief                To check print of the related information : relation, attribute name
+* @scenario             If used api returns DB_OK
+* @apicovered           db_print_header, db_query
+* @precondition         Database resource must be deinitialized
+* @postcondition        none
+*/
+void itc_arastorage_db_print_header_n_after_deinit(void)
+{
+	const char *rel_name = RELATION_NAME2;
+	db_insert(rel_name);
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/ta_tc/arastorage/itc/itc_arastorage_main.c]: Now it is simple to understand.

**Comment by seinfra** [comment]
Code Rule Check OK.

File: apps/examples/testcase/ta_tc/arastorage/itc/itc_arastorage_main.c (hunk 2 above)
Comment (Taejun-Kwon) [reply apps/examples/testcase/ta_tc/arastorage/itc/itc_arastorage_main.c]: I mean you can make one function for below.
memset(g_query, 0, QUERY_LENGTH);
snprintf(g_query, QUERY_LENGTH, "CREATE ATTRIBUTE %s DOMAIN long IN %s;", g_attribute_set[1], rel_name);
ret = db_exec(g_query);
if (ret != DB_OK) {
	error_exec(rel_name);
	return;
}

**Comment by arvinmittal** [comment]
@Taejun-Kwon 
Resolved merge conflicts and fix review comments in new commit.
New commit PR : https://github.com/Samsung/TizenRT/pull/1138

**Pull request closed without merging**